From 8d8c6df309ff5fde9dd4770abcd4ec5d2449b3ec Mon Sep 17 00:00:00 2001
From: Matteo Beccati <matteo@beccati.com>
Date: Fri, 20 Nov 2015 08:27:54 +0100
Subject: [PATCH] Fix h1 report 97073

Persistent XSS
--------------

Tengku Zahasman has reported via HackerOne that usernames were not properly
escaped when displayed in the audit trail widget of the dashboard upon login,
allowing persistent XSS attacks. An authenticated user with enough privileges
to create other users could exploit the vulnerability to access the
administrator account.

A CVE-ID has been requested, but not assigned yet.

CWE: CWE-79
CVSSv2: 5.6 (AV:N/AC:H/Au:S/C:C/I:P/A:N)
---
 lib/templates/admin/dashboard/audit.html | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/templates/admin/dashboard/audit.html b/lib/templates/admin/dashboard/audit.html
index 98ae782c4a..3aebbef3a5 100644
--- a/lib/templates/admin/dashboard/audit.html
+++ b/lib/templates/admin/dashboard/audit.html
@@ -30,7 +30,7 @@
             <li>
                 <div class="title">
                     <a target="_top" href="userlog-audit-detailed.php?auditId={$aValue.auditid}">
-                        {$aValue.desc}
+                        {$aValue.desc|escape}
                     </a>
                 </div>
                 <div>
